app/src/main/java/com/github/se/project/ui/profile/EditTutorProfile.kt

```diff
@@ -0,0 +1,221 @@
+          ?: return Text(
+              text = "No Profile selected. Should not happen.",
+              modifier = Modifier.testTag("editTutorNoProfile"))
+
+  val profileLanguages = remember { mutableStateListOf<Language>() }
+  val profileSubjects = remember { mutableStateListOf<Subject>() }
+
+  LaunchedEffect(profile) {
+    profileLanguages.clear()
+    profileLanguages.addAll(profile.languages)
+
+    profileSubjects.clear()
+    profileSubjects.addAll(profile.subjects)
+  }
+
+  val priceSliderValue = remember { mutableFloatStateOf(profile.price.toFloat()) }
+  val showError = remember { mutableStateOf(false) }
+
+  val academicLevel = remember { mutableStateOf(profile.academicLevel.name) }
+  val section = remember { mutableStateOf(profile.section.name) }
+  var expandedSection by remember { mutableStateOf(false) }
+  var expandedAcademicLevel by remember { mutableStateOf(false) }
+
+  val context = LocalContext.current
+
+  Scaffold(
+      topBar = {
+        IconButton(
+            onClick = { navigationActions.goBack() },
+            modifier = Modifier.testTag("editTutorProfileCloseButton")) {
+              Icon(
+                  imageVector = Icons.AutoMirrored.Default.ArrowBack,
+                  contentDescription = "Go Back")
+            }
+      },
+      content = { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(horizontal = 16.dp)
+                    .padding(paddingValues)
+                    .testTag("tutorInfoScreen"),
+            verticalArrangement = Arrangement.spacedBy(8.dp)) {
+              Text(
+                  text = "${profile.firstName} ${profile.lastName}",
+                  modifier =
+                      Modifier.testTag(
+                          "editTutorWelcomeText"), // .padding(vertical = 48.dp, horizontal =
+                  // 16.dp).testTag("welcomeText"),
+                  style = MaterialTheme.typography.headlineMedium,
+                  textAlign = TextAlign.Center)
+
+              Text(
+                  "Modify your profile information:",
+                  style = MaterialTheme.typography.titleMedium,
+                  modifier = Modifier.testTag("editTutorProfileInstructionText"))
+
+              Spacer(modifier = Modifier.height(16.dp))
+
+              Text(
+                  "Teaching languages:",
+                  style = MaterialTheme.typography.titleSmall,
+                  modifier = Modifier.testTag("editTutorProfileLanguageText"))
+              // Language Selection
+              LanguageSelector(profileLanguages)
+
+              Spacer(modifier = Modifier.height(16.dp))
+
+              // Subject Selection
+              Text(
+                  "Teaching subjects:",
+                  style = MaterialTheme.typography.titleSmall,
+                  modifier = Modifier.testTag("editTutorProfileSubjectText"))
+              SubjectSelector(null, profileSubjects, true)
```

Comment:
Good initiative, using the pre-existing LanguageSelector and SubjectSelector components